### ISSUE TITLE
Consider extracting PasswordCredential and FederatedCredential to separate specs?

### ISSUE THREAD

**Issue by lukewarlow** [w3c/webappsec-credential-management#255]
This has somewhat come up before in #221 but I think it's worth raising again.

With this specification being the core base to many other specifications, (e.g. [FedCM](https://w3c-fedid.github.io/FedCM/)) with at least [WebAuthn](https://w3c.github.io/webauthn/) being a W3C recommendation, it would be better if this specification was just the core API surface required by other specifications. This would probably allow it to proceed through the W3C standardisation process and become a Recommendation itself rather than being stuck as a working draft.

Moving PasswordCredential and FederatedCredential to a Credential Management Level 2 could allow for level 1 to be progressed? But they seem unlikely to get traction any time soon (only Chromium ever implemented them and FederatedCredentials are superseeded by FedCM iirc?) and splitting the core specification doesn't feel like the best approach here.

Moving both into their own specifications seems like the best path forwards. Something like https://specs.lukewarlow.dev/password-credentials/ and https://specs.lukewarlow.dev/federated-credentials/ with those sections removed from the core spec and the registry table updated to the new standalone specs follows the patttern that other Credential types follow.

The main questions are is this something the working group want to do? And if so where would be best to put these new specs (WICG or WebAppSec)?

**Comment by marcoscaceres** [comment]
I tend to agree this would be a good idea. The PasswordCredential and FederatedCredential are fairly stable so it should be easy to extract them. A the same time, we should work out if we want to keep them on the REC track... I think PasswordCredential only has a single implementation, right? ... and I'm bit confused what the relationship is between FedCM and `FederatedCredential`? Is there any?

**Comment by lukewarlow** [comment]
> ...so it should be easy to extract them

Yeah I managed to do it quite easily.

> only has a single implementation (and no commitment from anyone else right now), right?

PasswordCredential as you say only has 1 implementation (Chromium). I raised https://github.com/mozilla/standards-positions/issues/842 and https://github.com/WebKit/standards-positions/issues/220 to try and get positions on that part of the API. Fwiw I think it's a good API and should be implemented across the board. But potentially as you say it would need to move to a note or maybe move back to WICG to be further incubated?

> and I'm bit confused what the relationship is between FedCM and FederatedCredential? Is there any?

My understanding is that FedCM is designed to address a superset of what FederatedCredential did. FederatedCredential should probably be deprecated but from what I've read that's harder from a webcompat point of view than expected.

**Comment by lukewarlow** [comment]
What would the steps be to get this done? I think it would be good to clean this spec up so it's just the core that's used by other specs such as WebAuthn. Having PasswordCredential and the semi deprecated FederatedCredential is unfortunate.

**Comment by mikewest** [comment]
With my WG hat on, yes, I think we should pull out the bits that there's less agreement on into distinct documents (we did it before... see e.g. https://github.com/w3c/webappsec-credential-management/commit/a43865bd8aa9842dbc8d8e688d5668c087677eac). The work @lukewarlow did seems like exactly that thing: moving those docs to WICG and going through the adoption process seems reasonable to me, and should let us get the core spec onto a CR snapshot maintenance model which would be excellent IMO.

**Comment by lukewarlow** [comment]
I don't think the spec has updated since I made those documents so they should actually be ready to go to move (with some metadata updates obvs) https://github.com/lukewarlow/password-credentials and https://github.com/lukewarlow/federated-credentials are the repos I'm happy to do anything that's needed to transfer them to WICG?

**Comment by lukewarlow** [comment]
Have opened https://github.com/WICG/proposals/issues/227 to propose moving the Password Credential there (obviously will need sign off from the WG before that happens)

I'm not sure what the best move is with FederatedCredential? Moving to WICG could make sense but then it's not something that will ever progress so perhaps a note is better for that as a historical here's what it looked like but go implement FedCM instead?

**Comment by lukewarlow** [comment]
I'm not sure how to get this added to the agenda nor in fact the details for the calls so if someone could help on that front I'd be thankful

**Comment by alsoLut** [comment]
@lukewarlow @mikewest 

Hello, will the PasswordCredential specification also clarify the expected behaviour for the following call when multiple credentials are saved for the site?  

```js
navigator.credentials.get({ password: true, mediation: "silent" })
```
